service/s3/s3crypto/s3crypto_client.go

```diff
@@ -0,0 +1,217 @@
+package s3crypto
+
+// DR TOOLS TEST integration test bucket
+import (
+	"bytes"
+	"encoding/base64"
+	"io/ioutil"
+	"math/rand"
+
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+const (
+	ivSize  = 16
```

Comment:
Tthe IV size is dependent on the algorithm being chosen: ECB doesn't use an IV, and the standard AES-GCM IV size is 12 bytes. If S3 or the other AWS SDKs use different conventions, those differences are worth noting in comments because they will come up when under crypto review.